contracts/src/PBHVerifierImplV1.sol

```diff
@@ -0,0 +1,205 @@
+    //   This ensures that the contract can safely be disposed of after it is no longer used.
+    // - Carefully consider what data recovery options are presented as new functionality is added.
+    //   Care must be taken to ensure that a migration plan can exist for cases where upgrades
+    //   cannot recover from an issue or vulnerability.
+
+    ///////////////////////////////////////////////////////////////////////////////
+    ///                    !!!!! DATA: DO NOT REORDER !!!!!                     ///
+    ///////////////////////////////////////////////////////////////////////////////
+
+    // To ensure compatibility between upgrades, it is exceedingly important that no reordering of
+    // these variables takes place. If reordering happens, a storage clash will occur (effectively a
+    // memory safety error).
+
+    using ByteHasher for bytes;
+
+    ///////////////////////////////////////////////////////////////////////////////
+    ///                                  ERRORS                                ///
+    //////////////////////////////////////////////////////////////////////////////
+
+    /// @notice Thrown when attempting to reuse a nullifier
+    error InvalidNullifier();
+
+    ///////////////////////////////////////////////////////////////////////////////
+    ///                                  Events                                ///
+    //////////////////////////////////////////////////////////////////////////////
+
+    /// @notice Emitted once for each successful PBH verification.
+    ///
+    /// @param root The root of the Merkle tree that this proof is valid for.
+    /// @param sender The sender of this particular transaction or UserOp.
+    /// @param nonce Transaction/UserOp nonce.
+    /// @param callData Transaction/UserOp call data.
+    /// @param pbhExternalNullifier External nullifier encoding month, year, and a pbhNonce.
+    /// @param nullifierHash Nullifier hash for this semaphore proof.
+    /// @param proof The zero-knowledge proof that demonstrates the claimer is registered with World ID.
+    event PBH(
+        uint256 indexed root,
+        address indexed sender,
+        uint256  nonce,
+        bytes callData,
+        uint256 indexed pbhExternalNullifier,
+        uint256 nullifierHash,
+        uint256[8] proof
+    );
+
+    event PBHVerifierImplInitialized(IWorldIDGroups indexed worldId, uint8 indexed numPbhPerMonth);
+
+    ///////////////////////////////////////////////////////////////////////////////
+    ///                                  Vars                                  ///
+    //////////////////////////////////////////////////////////////////////////////
+
+    /// @dev The World ID group ID (always 1)
+    uint256 internal immutable GROUP_ID = 1;
+
+    /// @dev The World ID instance that will be used for verifying proofs
+    IWorldIDGroups internal worldId;
+
+    /// @dev Make this configurable
+    uint8 internal numPbhPerMonth;
```

Comment:
Set this as external. This seems like a useful value for third parties to query.